Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/model/ResolvedIdentifierContext.kt

```diff
@@ -0,0 +1,123 @@
+package org.cqframework.cql.cql2elm.model
+
+import java.util.*
+import org.hl7.elm.r1.CodeDef
+import org.hl7.elm.r1.CodeSystemDef
+import org.hl7.elm.r1.ConceptDef
+import org.hl7.elm.r1.ContextDef
+import org.hl7.elm.r1.Element
+import org.hl7.elm.r1.ExpressionDef
+import org.hl7.elm.r1.OperandDef
+import org.hl7.elm.r1.ParameterDef
+import org.hl7.elm.r1.TupleElementDefinition
+import org.hl7.elm.r1.ValueSetDef
+
+/**
+ * Context for resolved identifiers containing the identifier, the resolved element (if non-null) as
+ * well as the type of matching done to retrieve the element, whether case-sensitive or
+ * case-insensitive.
+ */
+class ResolvedIdentifierContext
+private constructor(
+    private val identifier: String,
+    private val element: Element?,
+    private val matchType: ResolvedIdentifierMatchType
+) {
+    private enum class ResolvedIdentifierMatchType {
+        EXACT,
+        CASE_INSENSITIVE
+    }
+
+    val exactMatchElement: Optional<Element>
+        get() {
+            if (isExactMatch) {
+                return Optional.ofNullable(element)
+            }
+
+            return Optional.empty()
+        }
+
+    private val isExactMatch: Boolean
+        get() = ResolvedIdentifierMatchType.EXACT == matchType
+
+    fun warnCaseInsensitiveIfApplicable(): Optional<String> {
```

Comment:
Why an Optional<String> and not a String?

Reply:
This was how to Kotlin convertor generated it. I'll take another look and see if it can be simplified.

Reply:
Went with `String?`, but it's about the same as far as code clarity. Will need to experiment more with `Optional`. In principle it's made redundant by having explicit nullability. I kind of like the `map` functionality though.